POI parsing and spawning

SloopyNoSleep opened the pull request:
Code is a bit icky, more time needs to be spent on conceptualizing how to handle mapping between a simple "encompassing" type that gets deserialized from json and how it has logic that spawns appropriate gameobject with transform, relationships, and components.

Also I can't solve everything with inheritance like the current typeconverter leans into. Makes composition of objects hard especially without a 1-to-1 mapping to internal types. I'll probably make more commits to try to clean this up a little (but I can make those in a future PR if we don't mind the technical debt 😝 )

SloopyNoSleep committed:
Add POI deserialization logic with stubs

SloopyNoSleep committed:
Enable Nullable and apply where applicable

SloopyNoSleep committed:
Merge remote-tracking branch 'upstream/master'

SloopyNoSleep committed:
Apply missing nullable adaptations post merge

SloopyNoSleep committed:
Use shortened new notation

SloopyNoSleep committed:
Add minmal CustomPoi representation and converter

SloopyNoSleep committed:
Update TextureUtil with new language features

SloopyNoSleep committed:
Add PoiUtil to help define and create POIs

SloopyNoSleep committed:
Add patch for populating object dicts and creating Pois

SloopyNoSleep committed:
Add missing Unity physics dependency

SloopyNoSleep committed:
Add example harvest poi def near gale cliffs merchant

SloopyNoSleep committed:
Add player patch for item population and poi spawning

I thought I added this earlier.... I suppose I wasn't paying attention

SloopyNoSleep committed:
Merge branch 'master' of https://github.com/Hacktix/Winch into poi_parsing

bdlm-dev commented on Winch/Serialization/DredgeTypeHelpers.cs:
```diff
@@ -15,20 +15,31 @@ public static class DredgeTypeHelpers
     {
         int r = 0, g = 0, b = 0, a = 255;
-        if (color.TryGetValue("r", out var value)) r = value;
+       
```
Red missing here

SloopyNoSleep replied:
Darn, that's what I get for copy-pasting. Nice catch, thanks!

bdlm-dev commented on Winch/Serialization/Item/SpatialItemDataConverter.cs:
```diff
@@ -26,7 +26,7 @@ public class SpatialItemDataConverter : ItemDataConverter
         { "isUnderlayItem", new(false, null)},
         { "forbidStorageTray", new(false, null)},
-        { "dimensions", new(new List<Vector2Int>(){new Vector2Int(1,1)}, o => DredgeTypeHelpers.ParseDimensions((JArray)o)) },
+        { "dimensions", new(new List<Vector2Int>(){new(1,1)}, o => DredgeTypeHelpers.ParseDimensions((JArray)o)) },
```
I believe this should be (0, 0)

SloopyNoSleep replied:
Yes, you are very much right. Thank you

SloopyNoSleep committed:
Address problems/suggestions in PR

Accidentally removed r from color parsing logic and also ammended
default dimensions to start at 0,0 instead of 1,1.

SloopyNoSleep committed:
Remove Debug statements and clutter

Hacktix commented:
LGTM for now, we can always adjust type conversion later, this is a solid base for new modded content :)
